### PR TITLE
Customer list datagrid

### LINKED ISSUE #3822: Apply datagrid to customers page
### Describe a proposed solution
Apply read-only datagrid on the customers page

### Acceptance Criteria 
- [ ] As a user i should see customers rendered within the table (grid)
- [ ] As a user i should be able to use all of the functionalities that datagrid has in the other views (sorting, d&d columns)
- [ ] As a user without `MANAGE_ORDERS` permission I shouldn't be able to see "No. of orders" column in both picker and datagrid
- [ ] As a user I should be able to search & filter customers
- [ ] As a user I should be able to save search as a preset

### PR DESCRIPTION
**Pull request by Droniu**
Closes #3822, #3828.

### Screenshots

Before            |  After
:-------------------------:|:-------------------------:
image |  image

### Pull Request Checklist

1. [x] This code contains UI changes
2. [x] All visible strings are translated with proper context including data-formatting
3. [x] Attributes `data-test-id` are added for new elements
4. [x] The changes are tested in Chrome/Firefox/Safari browsers and in light/dark mode
5. [x] Your code works with the latest stable version of the core
6. [x] I added changesets and [read good practices](/.changeset/README.md)

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [x] stable
2. [ ] app
3. [ ] attribute
4. [ ] category
5. [ ] collection
6. [x] customer
7. [ ] giftCard
8. [ ] homePage
9. [ ] login
10. [ ] menuNavigation
11. [ ] navigation
12. [ ] orders
13. [ ] pages
14. [ ] payments
15. [ ] permissions
16. [ ] plugins
17. [ ] productType
18. [ ] products
19. [ ] sales
20. [ ] shipping
21. [ ] translations
22. [ ] variants
23. [ ] vouchers

CONTAINERS=4

### CODE REVIEW

File: src/customers/components/CustomerListDeleteButton/CustomerListDeleteButton.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { Button, Tooltip, TrashBinIcon } from "@saleor/macaw-ui/next";
+import React, { forwardRef, ReactNode, useState } from "react";
+
+interface CustomerDeleteButtonProps {
+  onClick: () => void;
+  children: ReactNode;
+}
+
+export const CustomerListDeleteButton = forwardRef<
```
Comment (poulch) [inline src/customers/components/CustomerListDeleteButton/CustomerListDeleteButton.tsx:9]: suggestion: use `BulkDeleteButton` from components

File: src/customers/components/CustomerListPage/CustomerListPage.tsx [hunk 2]
```diff
@@ -6,54 +6,46 @@ import {
...
+    >,
+    SortPage<CustomerListUrlSortField> {
+  customers: Customers | undefined;
   selectedCustomerIds: string[];
+  onTabUpdate: (tabName: string) => void;
+  loading: boolean;
+  hasPresetsChange: () => boolean;
+  onSelectCustomerIds: (rows: number[], clearSelection: () => void) => void;
+  onCustomersDelete: () => void;
+  onTabDelete: (id: number) => void;
```
Comment (poulch) [inline src/customers/components/CustomerListPage/CustomerListPage.tsx:48]: suggestion: I'd avoid using old name like `onTabUpdate`, I've create special interface for filter presets
Here is example from my PR with datagrid on discount page
```
export interface SaleListPageProps
  extends PageListProps,
    FilterPagePropsWithPresets<SaleFilterKeys, SaleListFilterOpts>,
    SortPage<SaleListUrlSortField>,
    ChannelProps {
  sales: SaleFragment[];
  selectedSaleIds: string[];
  onSalesDelete: () => void;
  onSelectSaleIds: (rows: number[], clearSelection: () => void) => void;
  ```

**Review by poulch — CHANGES_REQUESTED** [review]

File: src/components/TableFilter/FilterTabs.tsx [hunk 3]
```diff
@@ -14,7 +14,7 @@ const useStyles = makeStyles(
 
 interface FilterTabsProps {
   children?: React.ReactNode;
-  currentTab: number;
+  currentTab: number | undefined;
```
[the remark this answers is not in the document]
Comment (Droniu) [reply src/components/TableFilter/FilterTabs.tsx:17]: It's required to pass strict null check on multiple files.

[the remark this answers is not in the document]
Comment (poulch) [reply src/components/TableFilter/FilterTabs.tsx:17]: But this is the old component that should be used. This component has been used in `FilterBar` that should be replaced by `ListFilters`. So you change dead code

File: src/customers/views/CustomerList/CustomerList.tsx [hunk 4]
```diff
@@ -217,14 +219,14 @@ export const CustomerList: React.FC<CustomerListProps> = ({ params }) => {
...
+        onSubmit={onPresetSave}
       />
       <DeleteFilterTabDialog
         open={params.action === "delete-search"}
         confirmButtonState="default"
         onClose={closeModal}
-        onSubmit={handleTabDelete}
-        tabName={maybe(() => tabs[currentTab - 1].name, "...")}
+        onSubmit={onPresetDelete}
+        tabName={presets?.[presetIdToDelete ?? 1 - 1]?.name ?? "..."}
```
Comment (poulch) [inline src/customers/views/CustomerList/CustomerList.tsx:229]: suggestion: maybe will be better to extract this to separate function like this:
```
  const getFilterPresetDeleteName = () => {
    if (!presetIdToDelete || !presets[presetIdToDelete - 1]) {
      return "...";
    }

    return presets[presetIdToDelete - 1].name;
  };
  ```

**Review by poulch — APPROVED** [review]
LGTM, gr8 job 💪🏻

File: src/customers/views/CustomerList/CustomerList.tsx (hunk 4 above)
Comment (Droniu) [reply src/customers/views/CustomerList/CustomerList.tsx:229]: I implemented it like this before - it was crashing in some edge cases

**Review by andrzejewsky — DISMISSED** [review]

**Review by poulch — DISMISSED** [review]

**Comment by wojteknowacki** [comment]
Delete icon visible after customer deletion
Zrzut ekranu 2023-07-19 o 14 45 07

**Review by krzysztofzuraw — APPROVED** [review]

**Review by poulch — APPROVED** [review]

**Review by wojteknowacki — APPROVED** [review]

**Pull request merged**